design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+All new types of functionality, from an end user wanting to install a simple application to an administrator wanting to install support for low level infrastructure types, are all treated the same.
+They have the same approach to security and access control and the artifacts that they install have the same scope of impact to the control plane.
+This one size fits all approach is not ideal though, the multiple scenarios and personas need finer grain control around security, access, and isolation.
+
+This design doc proposes a new approach for the Crossplane Stack Manager (SM) that will support a smaller set of security permissions and increased control for the installation of stacks.
+
+## Design Summary
+
+The key differences in this design update are in the **scope** of the permissions that stacks can be granted within the control plane.
+
+* **Cluster Stacks** will be given permissions to watch, access, and create resources at a **cluster level**, i.e. across **all namespaces**.
+* **Stacks** will be given permissions only within a **single namespace**.
+* All stacks will be granted permissions to a small subset of core Kubernetes resources as well as access to any CRDs they own and depend on.
+
+This means that stacks should not be able to directly affect any resources outside of the specific namespace they are installed into, while cluster stacks operate at a higher level of privilege and can affect the entire cluster.
+This is a familiar concept to the Kubernetes permission scoping model of `ClusterRole` and `Role`.
+
+### CRD Scoping
+
+To avoid a common point of confusion, let's review the available scoping of CRDs.
+Custom resource **instances** can be scoped to either a single namespace or to the entire cluster.
+It's important to note that this refers to the instances, not the CRD type definitions.
+From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition):
+
+> CustomResourceDefinitions themselves are non-namespaced and are available to all namespaces
+
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
+This has caused confusion in the past, so it's important to clear that up here.
+
+## Allowed Resource Access
+
+As previously stated, Crossplane is not intended to be a typical/full Kubernetes cluster to run arbitrary workloads.
+Once installed into the control plane, stacks will mostly target external resources and a known set of CRDs that they own and depend on.
+Therefore, we can lock down the allowed permissions for stacks to a subset of all available permissions in the Kubernetes API.
+
+The `rbac.yaml` file that allows a stack to specify the permissions it requires will be removed and no longer supported.
+Instead, the permissions for a stack can be considered **implicit**.
```

Comment:
That's probably fine, though I am thinking we may want to acknowledge somewhere that there is a usability tradeoff here. On one hand, hiding complexity makes things easier for the expected use-cases. On the other hand, it often makes things harder for the unexpected use-cases.

Related question: do we expect to support explicit configuration in the future?

Reply:
I feel that this conversation is resolved, so I'm going to resolve it.